### ISSUE TITLE
zip.compress does not close the dest file if src does not exist

### ISSUE THREAD

**Issue by mbertogliati** [deno-library/compress#19]
It seems that the zip.compress throws an exception if src does not exists, which seems correct, but it does not close the the dest (it shouldn't have opened it in the first place).

Minimal reproduction:
```
import { zip } from "jsr:@deno-library/compress";
import { describe, it} from "jsr:@std/testing/bdd";
import { expect } from "jsr:@std/expect";

describe("zip", () => {
  it("should throw if src does not exist", async () => {
    await expect(zip.compress("/tmp/non-existent.txt","/tmp/non-existent.zip")).rejects.toThrow();
  });
});
```
Logs:
```
running 1 test from ./main_test.ts
zip ...
  should throw if src does not exist ... ok (2ms)
zip ... FAILED (2ms)

 ERRORS 

zip => https://jsr.io/@std/testing/1.0.5/_test_suite.ts:218:10
error: Leaks detected:
  - A file was opened during the test, but not closed during the test. Close the file handle by calling `file.close()`.

 FAILURES 

zip => https://jsr.io/@std/testing/1.0.5/_test_suite.ts:218:10

FAILED | 0 passed (1 step) | 1 failed (5ms)

error: Test failed
```
I took a quick look into the code and i think that adding the line `await exists(src, { isFile: true });` at the beginning just like the `uncompress` function should do it. Maybe I'm wrong and its something else, I didn't look too much into it.

**Comment by fuxingZhang** [comment]
- need to judge the return value of `await exists(src, { isFile: true }); `
- src can be a file or a directory

**Comment by fuxingZhang** [comment]
`src` may be a folder

**Comment by fuxingZhang** [comment]
After testing, I found that Deno.stat will directly throw an error. Exists is used in many places but the return value is not judged.

**Comment by fuxingZhang** [comment]
fix: zip.compress does not close the dest file if src does not exist
```diff
export async function compress(
  src: string,
  dest: string,
  options?: compressInterface,
): Promise<void> {
+ const stat = await Deno.stat(src);
  const zipWriter = new ZipWriter((await Deno.create(dest)).writable);
  const inputs: Promise<EntryMetaData>[] = [];
- const stat = await Deno.stat(src);
```
fix: fix exists usage error
```diff
/**
 * Uncompresses a file.
 * @param {string} src - Source file path.
 * @param {string} dest - Destination file path.
 * @param {uncompressInterface} [options] - Optional parameters.
 */
export async function uncompress(
  src: string,
  dest: string,
  options?: uncompressInterface,
): Promise<void> {
-  await exists(src, { isFile: true });

+  const stat = await Deno.stat(src);
+  if(stat.isDirectory) {
+    throw new Error("The source path is a directory, not a file: ${src}")
+  }

-  for await (
-    const entry of (await Deno.open(src))
-      .readable
-      .pipeThrough(new ZipReaderStream())
-  ) {

+  using srcFile = await Deno.open(src);
+  for await (
+    const entry of srcFile.readable.pipeThrough(new ZipReaderStream())
+  ) {
    const filePath = path.resolve(dest, entry.filename);
    if (options?.debug) console.log(filePath);
    await Deno.mkdir(path.dirname(filePath), { recursive: true });
    if (entry.directory) continue;
    await entry.readable?.pipeTo((await Deno.create(filePath)).writable);
  }
  await terminateWorkers();
}
```

**Comment by fuxingZhang** [comment]
Landed on `0.5.5`

**Comment by fuxingZhang** [comment]
Landed on [`0.5.5`](https://jsr.io/@deno-library/compress)
